### PR TITLE
Fix Observation with unknown user in Observation index

### LINKED ISSUE #3609: Observation with nil User causes Error when shown in Index
114 Observations have a nil User.
```ruby
mushroom-observer(prod)> obss = Observation.where(user: nil)
=> 
[#<Observation:0x00007fa3f56480c0 ...
mushroom-observer(prod)> obss.size
=> 114
```
For more details about these Observations, see the [next comment](https://github.com/MushroomObserver/mushroom-observer/issues/3609#issuecomment-3661956666).

One issue is that displaying an Index page which includes any of these throws a Phlex Argument error.

To replicate: 
- Any search which returns an index of Observations of `Geastrum velutinum`
For instance: Observation pattern search for `Geastrum velutinum`
== http://localhost:3000/observations?q%5Bmodel%5D=Observation&q%5Bnames%5D%5Binclude_subtaxa%5D=true&q%5Bnames%5D%5Binclude_synonyms%5D=true&q%5Bnames%5D%5Blookup%5D%5B%5D=Geastrum+velutinum
Or this search which returns just two Observations, both of them bad 
http://localhost:3000/observations?q%5Bdate%5D%5B%5D=2016-03-18&q%5Bdate%5D%5B%5D=2016-03-18&q%5Bmodel%5D=Observation&q%5Bnames%5D%5Bexclude_consensus%5D=false&q%5Bnames%5D%5Binclude_all_name_proposals%5D=false&q%5Bnames%5D%5Binclude_subtaxa%5D=true&q%5Bnames%5D%5Binclude_synonyms%5D=false&q%5Bnames%5D%5Blookup%5D%5B%5D=Agaricus&q%5Bwithin_locations%5D%5B%5D=13403&q%5Bwithin_locations%5D%5B%5D=13402
- Result:
NOTE: Error report highlights `user_link` in line 157, `user_link(@obs.user)`
```ruby
Phlex::ArgumentError in Observations#index
Showing /Users/joe/mushroom-observer/app/views/controllers/observations/index.html.erb where line #19 raised:

You passed an unsafe object to `raw`.
Extracted source (around line #157):
              
  def render_obs_user(obs_user)
    if @user
      user_link(@obs.user)
    else
      plain(obs_user.unique_text_name)
    end

Rails.root: /Users/joe/mushroom-observer

Application Trace | Framework Trace | Full Trace
app/components/lightbox_caption.rb:157:in `render_obs_user'
app/components/lightbox_caption.rb:150:in `block in render_obs_who'
app/components/lightbox_caption.rb:148:in `render_obs_who'
app/components/lightbox_caption.rb:81:in `render_obs_when_where_who'
app/components/lightbox_caption.rb:51:in `render_obs_caption_parts'
app/components/lightbox_caption.rb:38:in `view_template'
app/components/base_image.rb:190:in `block in lightbox_caption_html'
app/components/base_image.rb:189:in `lightbox_caption_html'
app/components/base_image.rb:178:in `render_lightbox_link'
app/components/interactive_image.rb:86:in `render_image_overlays'
app/components/interactive_image.rb:46:in `block in render_interactive_image'
app/components/interactive_image.rb:39:in `render_interactive_image'
app/components/interactive_image.rb:33:in `view_template'
app/components/matrix_box.rb:80:in `block in render_thumbnail_section'
app/components/panel.rb:174:in `render_thumbnail'
app/components/panel.rb:76:in `block in <class:Panel>'
app/components/panel.rb:166:in `render_middle_sections'
app/components/panel.rb:159:in `block in render_thumbnail_and_body'
app/components/panel.rb:159:in `render_thumbnail_and_body'
app/components/panel.rb:95:in `block in view_template'
app/components/panel.rb:89:in `view_template'
app/components/matrix_box.rb:58:in `block in render_object_layout'
app/components/matrix_box.rb:54:in `render_object_layout'
app/components/matrix_box.rb:42:in `view_template'
app/components/matrix_table.rb:64:in `block (2 levels) in render_cached_boxes'
app/components/matrix_table.rb:63:in `block in render_cached_boxes'
app/components/matrix_table.rb:61:in `each'
app/components/matrix_table.rb:61:in `render_cached_boxes'
app/components/matrix_table.rb:49:in `block in view_template'
app/components/matrix_table.rb:39:in `view_template'
app/views/controllers/observations/index.html.erb:19
app/helpers/header/index_pagination_helper.rb:19:in `paginated_results'
app/views/controllers/observations/index.html.erb:18
app/controllers/application_controller/indexes.rb:263:in `show_index_of_objects'
app/controllers/application_controller/indexes.rb:198:in `filtered_index'
app/controllers/application_controller/indexes.rb:62:in `block in build_index_with_query'
app/controllers/application_controller/indexes.rb:49:in `each'
app/controllers/application_controller/indexes.rb:49:in `build_index_with_query'
app/controllers/observations_controller/index.rb:8:in `index'
app/controllers/application_controller.rb:148:in `catch_errors_and_log_request_stats'

Request
Parameters:
{"q"=>{"model"=>"Observation", "names"=>{"include_subtaxa"=>"true", "include_synonyms"=>"true", "lookup"=>["Geastrum velutinum"]}}}

Response
Headers:
None

```

### PR DESCRIPTION
**Pull request by JoeCohen**
When an observation's user is nil, the lightbox caption would fail with "Phlex::ArgumentError: You passed an unsafe object to raw".

Changes:
- Fix user_link helper to return html_safe string when user is nil
- Fix render_obs_user to consistently use obs_user parameter
- Add tests for user_link with nil and valid user

The bug occurred because user_link(nil) returned plain "?" string instead of an html_safe string, which Phlex requires for safe rendering.

# Manual Test

-  Run any Observation search whose first page of results includes >= 1 Obsevation whose user is nil
Examples:
  - Observation pattern search for Geastrum velutinum
== http://localhost:3000/observations?q%5Bmodel%5D=Observation&q%5Bnames%5D%5Binclude_subtaxa%5D=true&q%5Bnames%5D%5Binclude_synonyms%5D=true&q%5Bnames%5D%5Blookup%5D%5B%5D=Geastrum+velutinum
  - This search which returns just two Observations, both of them bad
http://localhost:3000/observations?q%5Bdate%5D%5B%5D=2016-03-18&q%5Bdate%5D%5B%5D=2016-03-18&q%5Bmodel%5D=Observation&q%5Bnames%5D%5Bexclude_consensus%5D=false&q%5Bnames%5D%5Binclude_all_name_proposals%5D=false&q%5Bnames%5D%5Binclude_subtaxa%5D=true&q%5Bnames%5D%5Binclude_synonyms%5D=false&q%5Bnames%5D%5Blookup%5D%5B%5D=Agaricus&q%5Bwithin_locations%5D%5B%5D=13403&q%5Bwithin_locations%5D%5B%5D=13402
- Expected result: The index displays without error.

### CODE REVIEW

File: app/helpers/object_link_helper.rb [hunk 1]
```diff
@@ -190,7 +190,7 @@ def wikipedia_term_search_url(name)
   #
   def user_link(user, name = nil, args = {})
     if !user
-      return "?"
+      return :unknown_user_name.t.html_safe # rubocop:disable Rails/OutputSafety
```
Comment (mo-nathan) [inline app/helpers/object_link_helper.rb]: I'm pretty sure the `.html_safe` here is redundant.
```
mushroom-observer(dev)> :unknown_user_name.t.class
=> ActiveSupport::SafeBuffer
mushroom-observer(dev)> :unknown_user_name.t.html_safe.class
=> ActiveSupport::SafeBuffer
```

**Review by mo-nathan — APPROVED** [review]
Overall looks good.  I'm pretty sure there is no reason at this point to do the `.html_safe` and I believe that is also what's requiring the rubocop disable.  I don't consider that a blocker so feel free to merge once you decide which way to go with it.
